File: app/src/main/java/com/karla/learningverbs/viewmodel/FavoritesViewModel.java
package com.karla.learningverbs.viewmodel;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.model.Verb;
import com.karla.learningverbs.repository.userrepository.UserRepository;
import com.karla.learningverbs.repository.verblist.VerbRepository;
import com.karla.learningverbs.utils.base.BaseViewModel;
import com.karla.learningverbs.utils.firebase.CustomListEventListener;

import java.util.ArrayList;
import java.util.List;

public class FavoritesViewModel extends BaseViewModel {
    private MutableLiveData<List<Verb>> getResultListFavoriteVerbs = new MutableLiveData<>();
    private VerbRepository favoritesRepository;
    private UserRepository userRepository;

    public FavoritesViewModel() {
        favoritesRepository = VerbRepository.getInstance();
        userRepository = UserRepository.getInstance();
    }

    public void getFavoriteVerbListElement() {
        favoritesRepository.getFavoritesListUser(userRepository.getUserId(), new CustomListEventListener<Verb>(Verb.class) {
            @Override
            public void onSuccess(ArrayList<Verb> response) {
                getResultListFavoriteVerbs.postValue(response);

            }

            @Override
            public void onFailed(Throwable throwable) {
                getResultListFavoriteVerbs.postValue(null);
            }

            @Override
            public void showLoaging() {
                loading.postValue(true);
            }

            @Override
            public void hideLoading() {
                loading.postValue(false);
            }
        });
    }

    public LiveData<List<Verb>> getListResultsFavoriteVerbs() {
        return getResultListFavoriteVerbs;
    }

}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/FavoritesViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.karla.learningverbs.kotlin.repository.userrepository.UserRepository
import com.karla.learningverbs.kotlin.repository.verblist.VerbRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel
import com.karla.learningverbs.kotlin.utils.firebase.CustomListEventListener
import com.karla.learningverbs.model.Verb

class FavoritesViewModel: BaseViewModel()  {
    private val getResultListFavoriteVerbs = MutableLiveData<ArrayList<Verb?>>()
    private var favoritesRepository: VerbRepository = VerbRepository()
    private var userRepository2: UserRepository = UserRepository()

    fun getFavoriteVerbListElement() {
        favoritesRepository.getFavoritesListUser(
            userRepository2.getUserId(),
            object : CustomListEventListener<Verb>(Verb::class.java) {
                override fun onSuccess(response: ArrayList<Verb?>?) {
                    getResultListFavoriteVerbs.postValue(response!!)
                }

                override fun onFailed(throwable: Throwable?) {
                    //getResultListFavoriteVerbs.postValue(null)
                    msgError.postValue(throwable!!.message)

                }

                override fun showLoaging() {
                    loading.postValue(true)
                }

                override fun hideLoading() {
                    loading.postValue(false)
                }
            })
    }

    fun getListResultsFavoriteVerbs(): LiveData<ArrayList<Verb?>> {
        return getResultListFavoriteVerbs
    }

}